src/service/fileSelectionService.ts

```diff
@@ -0,0 +1,94 @@
+
+export class FileSelectionService {
+    private currentFileAbsPath: string = "";
+    private currentTextDocument: vscode.TextDocument;
+    private messagingService: MessagingService;
+
+    constructor(messagingService: MessagingService) {
+        this.messagingService = messagingService;
+    }
+
+    public getCurrentFileAbsPath() {
+        return this.currentFileAbsPath;
+    }
+    public getCurrentTextDocument() {
+        return this.currentTextDocument;
+    }
+    public updateCurrentFileFromTextFile = async (
+        activeTextDocument: vscode.TextDocument | undefined
+    ) => {
+        if (activeTextDocument) {
+            await this.updateCurrentFileFromEditor({
+                document: activeTextDocument,
+            } as vscode.TextEditor);
+        } else {
+            return;
+        }
+    };
+    public updateCurrentFileFromEditor = async (
+        activeTextDocument: vscode.TextEditor | undefined
+    ) => {
+        if (
+            activeTextDocument &&
+            activeTextDocument.document &&
+            activeTextDocument.document.languageId === "python"
+        ) {
+            this.setPathAndSendMessage(activeTextDocument.document.fileName);
+            this.currentTextDocument = activeTextDocument.document;
+        } else if (this.currentFileAbsPath === "") {
+            this.setPathAndSendMessage(this.getActivePythonFile() || "");
+        }
+        if (
+            this.currentTextDocument &&
+            this.getActiveEditorFromPath(this.currentTextDocument.fileName) ===
+                undefined
+        ) {
+            await vscode.window.showTextDocument(
+                this.currentTextDocument,
+                vscode.ViewColumn.One
+            );
+        }
+    };
+    public findCurrentTextDocument() {
+        if (this.currentFileAbsPath) {
+            const foundDocument = this.getActiveEditorFromPath(
+                this.currentFileAbsPath
+            );
+            if (foundDocument !== undefined) {
+                this.currentTextDocument = foundDocument;
+            }
+        }
+    }
+
+    public setPathAndSendMessage = (newFilePath: string) => {
+        this.currentFileAbsPath = newFilePath;
+        this.messagingService.sendMessageToWebview(
+            VSCODE_MESSAGES_TO_WEBVIEW.CURRENT_FILE,
+            {
+                running_file: newFilePath,
+            }
+        );
+    };
+    private getActiveEditorFromPath = (
+        filePath: string
+    ): vscode.TextDocument => {
+        const activeEditor = vscode.window.visibleTextEditors.find(
+            (editor: vscode.TextEditor) => editor.document.fileName === filePath
+        );
+        return activeEditor ? activeEditor.document : undefined;
+    };
+    private getActivePythonFile = () => {
+        const editors: vscode.TextEditor[] = vscode.window.visibleTextEditors;
+        const activeEditor = editors.find(
+            editor => editor.document.languageId === "python"

```

Comment:
Consider making "python" a constant for these identifiers? We already have one defined at `GLOBAL_ENV_VARS.PYTHON`, but maybe we should add a new one specifically for VS Code language IDs... Anyway, it's up to you!

Reply:
done!